annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+        task_data, all_tasks
+    )
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
+
+
+def test_find_common_objs_no_common_objects():
+    task_data = {
+        "objects": [{"id": 3, "name": "Object D", "value": "Unique Value"}]
+    }
+
+    all_tasks = {
+        1: [({"name": "Object A", "value": "Some Value"}, "1")],
+        2: [({"name": "Object B", "value": "Another Value"}, "2")],
+    }
+
+    expected_common_objs = []
+    expected_same_ids = {}
+
+    result_common_objs, result_same_ids = find_common_objs(
+        task_data, all_tasks
+    )
+
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
+
+
+def test_find_common_objs_empty_data():
+    task_data = {"objects": []}
+
+    all_tasks = {}
+
+    expected_common_objs = []
+    expected_same_ids = {}
+
+    result_common_objs, result_same_ids = find_common_objs(
+        task_data, all_tasks
+    )
+
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
+
+
+@pytest.mark.parametrize(
+    ("tasks", "expected"),
+    (
+        ({}, {}),
+        (
+            {
+                1: {
+                    "objects": [
+                        {
+                            "id": "1",
+                            "name": "Task 1",
+                            "details": {"id": "a1", "info": "Details 1"},
+                        },
+                        {"id": "2", "name": "Task 2"},
+                    ]
+                }
+            },
+            {
+                1: [
+                    (
+                        {
+                            "name": "Task 1",
+                            "details": {"id": "a1", "info": "Details 1"},
+                        },
+                        "1",
+                    ),
+                    ({"name": "Task 2"}, "2"),
+                ]
+            },
+        ),
+        (
+            {
+                2: {
+                    "objects": [
+                        {
+                            "id": "3",
+                            "info": {"id": "b3", "value": "Nested Value"},
+                        },
+                    ]
+                }
+            },
+            {2: [({"info": {"id": "b3", "value": "Nested Value"}}, "3")]},
+        ),
+        (
+            {
+                3: {
+                    "objects": [
+                        {"id": "4", "name": "Task 4"},
+                        {"id": "5", "status": "Complete"},
+                    ]
+                },
+                4: {
+                    "objects": [
+                        {"id": "6", "description": "Task 6"},
+                    ]
+                },
+            },
+            {
+                3: [({"name": "Task 4"}, "4"), ({"status": "Complete"}, "5")],
+                4: [({"description": "Task 6"}, "6")],
+            },
+        ),
+    ),
+)
```

Comment:
These parametrized tests quickly get unreadable. Could you create shorter parameters, with less unnecessary attributes? In addition, I think in the 3rd test the 4:{...} is already covered as a case.